app/src/main/java/com/nervousfish/nervousfish/activities/SettingsActivity.java

```diff
@@ -289,4 +285,72 @@ public boolean onOptionsItemSelected(final MenuItem item) {
             return super.onOptionsItemSelected(item);
         }
     }
+
+    private static final class BindPreferenceToValueListener implements Preference.OnPreferenceChangeListener {
+        @Override
+        public boolean onPreferenceChange(final Preference preference, final Object newValue) {
+            LOGGER.info("Preference changed");
+            final String stringValue = newValue.toString();
+
+            if (preference.getKey().equals(ConstantKeywords.DISPLAY_NAME)) {
+                LOGGER.info("Preference changed at the display name");
+                updateDisplayName(preference, stringValue);
+                return true;
+            } else if (preference instanceof ListPreference) {
+                LOGGER.info("Preference changed for a ListPreference");
+                updateListPreference(preference, stringValue);
+                return true;
+            } else {
+                LOGGER.info("Preference changed which is not a ListPreference, and not the display name");
+                // For all other preferences, set the summary to the value's
+                // simple string representation.
+                preference.setSummary(stringValue);
+                return true;
+            }
+        }
+
+        /**
+         * If the key is display_name this method is called to update the summary
+         * and the Profile in the database.
+         *
+         * @param preference  The preference which is changed
+         * @param stringValue The string value which is new
+         */
+        private static void updateDisplayName(final Preference preference, final String stringValue) {
+            if (firstLoad) {
+                firstLoad = false;
+                preference.setSummary(serviceLocator.getDatabase().getProfile().getName());
+            } else {
+                try {
+                    LOGGER.info("Updating profile name");
+                    final Profile profile = serviceLocator.getDatabase().getProfile();
+                    serviceLocator.getDatabase().updateProfile(new Profile(stringValue, profile.getKeyPairs()));
+                } catch (final IOException e) {
+                    LOGGER.error("Couldn't get profiles from database", e);
+                }
+
+                preference.setSummary(stringValue);
+            }
+        }
+
+        /**
+         * If the preference is a list preference this method is called to update the summary.
```

Comment:
Either remove the dot at the endof this line or add one in the method above this one 🙈